### PR TITLE
Update custom user model to adjust for changes in extensions

### PR DESCRIPTION
**Pull request by rlucioni**
Also includes clean up of settings, README, and dependencies. Accompanies PR [#24](https://github.com/edx/django-oscar-extensions/pull/24) against django-oscar-extensions. @clintonb, could you take a quick look when you have a minute?

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -2,6 +2,7 @@
 [run]
 data_file = .coverage
 omit = ecommerce/settings*
+    ecommerce/user*
```
Comment (clintonb) [inline .coveragerc]: This app should still be represented in coverage reports. There is custom code there that should be tested.

**Comment by clintonb** [comment]
Migrations are missing. Otherwise, this looks good.

File: Makefile [hunk 2]
```diff
@@ -21,9 +20,6 @@ dev_requirements:
 test_requirements:
 	pip install -qr requirements/test.txt --exists-action w
 
-super:
```
Comment (clintonb) [inline Makefile:24]: Why was this removed?

File: .coveragerc (hunk 1 above)
Comment (rlucioni) [inline .coveragerc]: The custom code is covered, but something about the way Django 1.7 loads models causes the rest of the app to [appear uncovered](https://i.imgur.com/IPOAUnI.png). The result is skewed coverage stats. I'll remove this line if you're okay with 88% coverage :cry: .

File: Makefile (hunk 2 above)
Comment (rlucioni) [inline Makefile:24]: The ecommerce service uses the LMS to authenticate and create users with their LMS permissions preserved. It doesn't seem to make much sense to encourage superuser creation by supplying a `make` target for it if you won't be able to log in as the superuser you created. If you need a superuser for the ecommerce service, you'll just log in as a superuser from the LMS.

**Comment by rlucioni** [comment]
@clintonb could you elaborate on the missing migrations? The `health` app is unmigrated, and the service hasn't been deployed yet, so it doesn't seem like any changes contained here require migrations.

File: .coveragerc (hunk 1 above)
Comment (clintonb) [inline .coveragerc]: Ah. I see. Keep it, but add a note stating why this line is there.

File: Makefile (hunk 2 above)
Comment (clintonb) [inline Makefile:24]: Makes sense.

File: .coveragerc (hunk 1 above)
Comment (rlucioni) [inline .coveragerc]: Sounds good.

**Comment by rlucioni** [comment]
@clintonb I've added the missing migrations to create the user table; thanks for catching that.

**Comment by clintonb** [comment]
:+1:
